src/Microsoft.AspNet.Mvc.Localization/HtmlLocalizer.cs

```diff
@@ -0,0 +1,233 @@
+
+        /// <summary>
+        /// Encodes the arguments based on the object type.
+        /// </summary>
+        /// <param name="resourceString">The resourceString whose arguments need to be encoded.</param>
+        /// <param name="arguments">The array of objects to encode.</param>
+        /// <returns>The string with encoded arguments.</returns>
+        protected virtual string EncodeArguments([NotNull] string resourceString, [NotNull] object[] arguments)
+        {
+            var position = 0;
+            var length = resourceString.Length;
+            char currentCharacter;
+            StringBuilder tokenBuffer = null;
+            var outputBuffer = new StringBuilder();
+            var isToken = false;
+            var isEncode = true;
+
+            while (position < length)
+            {
+                currentCharacter = resourceString[position];
+
+                position++;
+                if (currentCharacter == '}')
+                {
+                    if (position < length && resourceString[position] == '}')  // Treat as escape character for }}
+                    {
+                        if (isToken)
+                        {
+                            isEncode = false;
+                            tokenBuffer.Append("}}");
+                        }
+                        else
+                        {
+                            outputBuffer.Append("}");
+                        }
+
+                        position++;
+                        if (position == length)
+                        {
+                            AppendToOutputBuffer(isEncode, arguments, tokenBuffer, ref outputBuffer);
+                            break;
+                        }
+                    }
+                    else
+                    {
+                        AppendToBuffer(isToken, "}", ref tokenBuffer, ref outputBuffer);
+
+                        AppendToOutputBuffer(isEncode, arguments, tokenBuffer, ref outputBuffer);
+
+                        if (position == length)
+                        {
+                            break;
+                        }
+
+                        isToken = false;
+                        tokenBuffer = null;
+                        isEncode = true;
+                    }
+                }
+                else if (currentCharacter == '{')
+                {
+                    if (position < length && resourceString[position] == '{')  // Treat as escape character for {{
+                    {
+                        if (isToken)
+                        {
+                            tokenBuffer.Append("{{");
+                            position++;
+                        }
+                        else
+                        {
+                            position++;
+                            if (position < length &&
+                                (resourceString[position] > '0' && resourceString[position] < '9'))
```

Comment:
Shouldn't the "{{" case look more like the "}}" case? What is this condition and the positive block for? What tests execute the `tokenBuffer = new StringBuilder(); ...` bit?

If needed, why aren't the comparisons `>=` and `<=`?